### PR TITLE
Added more samples to exemplar

### PR DESCRIPTION
**Pull request by jeremyholtzman**
## Description & motivation

Tasks can be found here - https://app.asana.com/0/1204502909052868/1204883474581691/f

The PR includes:
- more incremental examples with new data source
- more snapshot examples
- tests on snapshots
- how to call the API to run a job
- custom selector
- node slection syntax examples
- added in dbt_artifacts package
- moved dim_customers_legacy to analysis folder

## Screenshots (DAG, query results):

Screenshot 2023-06-27 at 4 07 37 PM
Screenshot 2023-06-27 at 4 09 48 PM

### CODE REVIEW

**Comment by jeremyholtzman** [comment]
Hey @MichelleBaird-DBT, could you give a review to this PR which adds a bunch of examples to the exemplar project?

Things you should specifically look for:

- make sure all of the examples added in the analysis folder make sense and test them out yourself
- review the concepts in all of the new files I added to make sure they would be adequate for demo-ing if needed
- any notes on naming convention and project organization

I tested that the project runs properly, but feel free to give it a go as well.

**Comment by MichelleBaird-DBT** [comment]
I think this should be good to go! Naming convention and organization look logical to me. Examples make sense to include, should be good to demo.

File: models/examples/model_governance/example_private_finance_model_v2.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+select 
+    * exclude priority_code
```
Comment (bethanyhipple-dbtlabs) [inline models/examples/model_governance/example_private_finance_model_v2.sql:2]: WHAT??? I didn't know you could do a select * and exclude a column. super cool!

File: selectors.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+selectors:
+  - name: all_models_and_tagged_snapshots
+    description: includes all models, but only snapshots any snapshot tagged daily
```
Comment (bethanyhipple-dbtlabs) [inline selectors.yml:3]:
```suggestion
    description: includes all models, but only snapshots tagged daily
```

**Review by bethanyhipple-dbtlabs — APPROVED** [review]
Only one suggestion with the description on 'models and tagged snapshots'

File: models/examples/model_governance/example_private_finance_model_v2.sql (hunk 1 above)
Comment (jeremyholtzman) [reply models/examples/model_governance/example_private_finance_model_v2.sql:2]: yeah it's a snowflake thing! Idk when they released it but it's the first time I used it :)

**Pull request merged**